src/adapter/mod.rs

```diff
@@ -52,6 +52,13 @@ impl<'a> RustdocAdapter<'a> {
     pub fn schema() -> &'static Schema {
         &SCHEMA
     }
+
+    pub(crate) fn crate_at_origin(&self, origin: &Origin) -> Option<&'a PackageIndex<'a>> {
```

Comment:
I think `Origin` might be `Copy` so we might not need `&Origin`, `Origin` would do.